src/dmclock/CMakeLists.txt

```diff
@@ -16,10 +16,12 @@ if(K_WAY_HEAP)
   endif()
 endif()
 
-if (NOT(TARGET gtest AND TARGET gtest_main))
-  find_package(gtest REQUIRED)
-  include_directories(${GTEST_INCLUDE_DIRS})
-endif()
+if(WITH_DMCLOCK_TESTS)
```

Comment:
WITH_DMCLOCK_TESTS should not appear below src/dmclock. A better approach is the one used in https://github.com/ceph/ceph/pull/14997. In fact a good starting point might be the final two commits in that PR. The first is a git subtree pull of the latest dmclock master that significantly reorganizes dmclock's cmake process (see README.git-subtree to see how this is done). The second (commit 8215686) are corresponding changes in ceph's cmake process. This approach as a starting point will avoid conflicts and re-visiting this issue down the line.